IDEAS/Buildings/Components/RectangularZoneTemplate.mo

```diff
@@ -333,8 +334,9 @@ protected
     fraType(present=fraTypA.present,
             U_value=fraTypA.U_value),
     linExtRad=linExtRadWin) if
-       hasWinA
+       hasWinA constrainedby Window
```

Comment:
the parameters
```
      inc=IDEAS.Types.Tilt.Wall,
    glazing(
      nLay=glazingB.nLay,
      mats=glazingB.mats,
      SwAbs=glazingB.SwAbs,
      SwTrans=glazingB.SwTrans,
      SwAbsDif=glazingB.SwAbsDif,
      SwTransDif=glazingB.SwTransDif,
      U_value=glazingB.U_value,
      g_value=glazingB.g_value),
    A=A_winB,
    frac=fraB,
    azi=aziA + Modelica.Constants.pi/2,
    T_start=T_start,
    linIntCon_a=linIntCon,
    dT_nominal_a=dT_nominal_win,
    linExtCon=linExtCon,
    windowDynamicsType=windowDynamicsType,
```
etc have to be in the `constrainedby` clause when making a component replaceable. Otherwise the parameter values can be 'forgotten' when redeclaring a component. This depends on the Modelica tool that is used, but better be safe than sorry!